internal/icinga2/util.go

```diff
@@ -18,6 +19,23 @@ func rawurlencode(s string) string {
 }
 
 // isMuted returns true if the given checkable is either in Downtime, Flapping or acknowledged, otherwise false.
-func isMuted(checkable *ObjectQueriesResult[HostServiceRuntimeAttributes]) bool {
-	return checkable.Attrs.IsFlapping || checkable.Attrs.Acknowledgement != AcknowledgementNone || checkable.Attrs.DowntimeDepth != 0
+// When the checkable is Flapping, and neither the flapping detection for that Checkable nor for the entire zone is
+// enabled, this will always return false.
+func isMuted(ctx context.Context, client *Client, checkable *ObjectQueriesResult[HostServiceRuntimeAttributes]) (bool, error) {
+	if checkable.Attrs.IsFlapping {
+		if checkable.Attrs.EnableFlapping {
+			return true, nil
+		}
+
+		status, err := client.fetchIcingaAppStatus(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		if status.App.EnableFlapping {
+			return true, nil
+		}
```

Comment:
Have you verified that this actually matches the logic in Icinga 2? Like does the checkable attribute really have precedence over the global attribute? My intuition is that it has to be enabled both globally and on the checkable to actually be active, i.e. the global setting is more like a kill switch.